Integrating the notification service

ebenezergraham opened the pull request:
1 Added the repository to the initial-setup and pull-all scripts
2 Added notification service dependency to the Gradle file
3 Created an instance of the notification manager
4 Wrote the start and tear down methods
5 Created a listener class for the notification service
6 Added a notification role and gave it customer read permission
7 Created a notification admin user for notification operations

 File changes to be committed:
        modified:   build.gradle
        modified:   scripts/Unix/initial-setup.sh
        modified:   scripts/Unix/pull-all.sh
        modified:   scripts/windows/initial-setup.bat
        modified:   scripts/windows/pull-all.bat
        modified:   src/main/java/org/apache/fineract/cn/dev/ServiceRunner.java
        new file:   src/main/java/org/apache/fineract/cn/dev/listener/NotificationListener.java

myrle-krantz commented:
Looks good.  I'd only like to ask for one change before I merge:

Please give the notification user and the password constants more specific names.  e.g. USER_IDENTIFIER -> NOTIFICATION_USER_IDENTIFIER

Once you improve the application authentication code for accessing customer, you'll probably want to update this code again.  But let's get this bit merged now anyways.

ebenezergraham committed:
1 Added the repository to the initial-setup and pull-all scripts

2 Added notification service dependency to the gradle file
3 Created an instance of the notificationManager
4 Wrote the start and tear down methods
5 Created a listener class for the notificaiton service
6 Added a notification role and gave it customer read permission
7 Created a notification admin user for notification operations

 File changes to be committed:
        modified:   build.gradle
        modified:   scripts/Unix/initial-setup.sh
        modified:   scripts/Unix/pull-all.sh
        modified:   scripts/windows/initial-setup.bat
        modified:   scripts/windows/pull-all.bat
        modified:   src/main/java/org/apache/fineract/cn/dev/ServiceRunner.java
        new file:   src/main/java/org/apache/fineract/cn/dev/listener/NotificationListener.java

ebenezergraham commented:
@myrle-krantz I have changed the constant name and squashed it. Loving this squash feature :smile:

myrle-krantz commented:
Looks good.  Thank you very much for the PR Ebenezer!
